### PR TITLE
Added support for editing mods

### PR DESCRIPTION
**Pull request by Auros**

### CODE REVIEW

**Review by Caeden117 — APPROVED** [review]
LGTM

File: src/Hive/Services/Common/ModService.cs [hunk 1]
```diff
@@ -271,6 +275,66 @@ public async Task<HiveObjectQuery<Mod>> MoveMod(User? user, string channelDestin
...
+                return new HiveObjectQuery<Mod>(StatusCodes.Status400BadRequest, "Mod Update is invalid");
+
+            var targetVersion = new Version(update.Version);
+
+            // Get the database mod that represents the SerializedModUpdate.
+            var databaseMod = await CreateModQuery().Where(x => x.ReadableID == update.ID && x.Version == targetVersion)
+                .FirstOrDefaultAsync().ConfigureAwait(false);
+
+            // The POSTed mod was successfully deserialzed, but no Mod exists in the database.
+            if (databaseMod == null)
```
Comment (sc2ad) [inline src/Hive/Services/Common/ModService.cs:298]: But: is null

Comment (sc2ad) [reply src/Hive/Services/Common/ModService.cs:298]: Nit*

**Review by sc2ad — APPROVED** [review]
I think we should revisit permission differences (or if we even want permission differences) for this endpoint after some time. Lgtm for this though, test is clear enough but I wonder what happens if you remove things that would otherwise have been tracked (ex, game versions that were supported but no longer are, etc.)

File: src/Hive/Services/Common/ModService.cs (hunk 1 above)
Comment (Auros) [reply src/Hive/Services/Common/ModService.cs:298]: I copy pasted from the [MoveMod method](https://github.com/Atlas-Rhythm/Hive/blob/20d99577074f6107c411210cfea9ba793f38e65d/src/Hive/Services/Common/ModService.cs#L241) haha

**Pull request merged**